### PR TITLE
Fix "No object found for UID: <laboratory_uid>" in report preview

### PR DESCRIPTION
**Pull request by xispa**
## Description of the issue/feature this PR addresses

This Pull Request fixes a "No object found for UID" error in results report preview. The underlying cause if this error is that somehow, the laboratory content type is no longer indexed in the database, probably due to some catalog migrations done recently

## Current behavior before PR

```
Error: No object found for UID '8f471eec8ab14aaaaff775270ab483c3'

 - Expression: "laboratory/title|nothing"
 
 ```

## Desired behavior after PR is merged

No error, the report renders correctly

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

File: src/senaite/core/upgrade/v02_03_000.py [hunk 1]
```diff
@@ -420,3 +426,49 @@ def fixed_point_value_to_string(value, precision):
     str_value = template % (sign, front, fra)
     # strip off trailing zeros and possible dot
     return str_value.rstrip("0").rstrip(".")
+
+
+def setup_catalogs_order(portal):
```
Comment (ramonski) [inline src/senaite/core/upgrade/v02_03_000.py]: Would you mind to add this as a setuphandler step and use it from there?

File: src/senaite/core/upgrade/v02_03_000.py [hunk 2]
```diff
@@ -420,3 +426,49 @@ def fixed_point_value_to_string(value, precision):
...
+        if id1 == id2:
+            return 0
+
+        # Catalogs sorted, senaite_* always first
+        senaite = map(lambda cat_id: cat_id.startswith("senaite_"), [id1, id2])
+        if not all(senaite) and any(senaite):
+            # Item starting with senaite always gets max priority
+            if id1.startswith("senaite_"):
+                return -1
+            return 1
```
Comment (ramonski) [inline src/senaite/core/upgrade/v02_03_000.py]: What do you think about a list of all catalogs, e.g. `[SETUP_CATALOG, SAMPLE_CATALOG ... AUDITLOG_CATALOG]` and use the index as sort key?

**Review by ramonski — CHANGES_REQUESTED** [review]

File: src/senaite/core/upgrade/v02_03_000.py (hunk 1 above)
Comment (xispa) [reply src/senaite/core/upgrade/v02_03_000.py]: Sure

File: src/senaite/core/upgrade/v02_03_000.py (hunk 2 above)
Comment (xispa) [reply src/senaite/core/upgrade/v02_03_000.py]: I think is not necessary for now, keep in mind that other extension profiles might add new catalogs starting with `senaite_` prefix as well

File: src/senaite/core/upgrade/v02_03_000.py (hunk 1 above)
Comment (xispa) [reply src/senaite/core/upgrade/v02_03_000.py]: Done with https://github.com/senaite/senaite.core/pull/2146/commits/70b363d487e34653e137c3013f7ced38cc822cc2

**Review by ramonski — APPROVED** [review]

**Pull request merged**